### PR TITLE
Update encyclopedia to v2

### PR DESCRIPTION
**Pull request by jspaezp**
https://github.com/TalusBio/nf-encyclopedia/issues/39

### CODE REVIEW

**Comment by jspaezp** [comment]
tests passed in https://github.com/TalusBio/nf-encyclopedia/actions/runs/5247419174 but I added some updates to the actions versions to remove some of the warnings.

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,18 @@
+FROM --platform=linux/amd64 mambaorg/micromamba:latest as micromamba
+# First Stage of the build, gets the jar for encyclopedia
+FROM --platform=linux/amd64 openjdk:8-jre
 ARG VERSION=1.12.34
```
Comment (wfondrie) [inline Dockerfile:4]: **Question:** We need to bump this to 2.x right?

**Review by wfondrie — APPROVED** [review]
This looks great. I approved, but left one question.

File: Dockerfile (hunk 1 above)
Comment (jspaezp) [reply Dockerfile:4]: I cannot believe I did not change this ...

Comment (jspaezp) [reply Dockerfile]: changed in cce4906363f095402c8795da5f3205ead75652e9

**Pull request merged**